threatmodels/notation-theatmodel.md

```diff
@@ -0,0 +1,85 @@
+- **File System**: The file system where Notation or application built with Notation library is installed. For Notation signing local artifact, it is the place where local artifact and associated signatures are stored. There are different stores as follows
+  - Notation config store, where Notation configuration file are stored, for example, trust policy configuration
+  - Notation trust store, where certificates are added to validate signatures
+  - Notation plugin store, where Notation plugins are installed
+  - Credential store, where the registry or Key management credentials are stored
+- **Notation Plugin**: An external component built by different vendors, and integrated to Notation as a signing plugin for signature generation, or as a verifying plugin for signature verification, see [plugin spec](https://github.com/notaryproject/notaryproject/blob/v1.0.0-rc.2/specs/plugin-extensibility.md) for details.
+- **Registry**: A remote registry of Docker and Open Container Initiative (OCI) images, with support for all OCI artifacts. It is the remote storage where artifacts are pushed to by artifact builder. Registries are outside Notation trust boundary.
+- **KMS**: A remote Key Management Service provides key and associated certificate management, and remote signing service. KMSs are outside Notation trust boundary.
+
+## Notation sign artifacts using remote key
+
+### Data flow
+
+A signer uses Notation to sign artifacts with the private key stored in a KMS, and store signatures in a local file system or in a remote registry.
+
+The following diagram illustrates Notation signing remote artifacts with KMS:
+
+![image](../media/notation-sign-remote.png)
+
+The following diagram illustrates Notation signing local artifacts with KMS:
+
+![image](../media/notation-sign-local.png)
+
+According to [Notary signature specification](../specs/signature-specification.md), the signing payload is a JSON document which requires the [OCI descriptor](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/descriptor.md) of the artifact. For artifacts stored in a registry, the descriptor info can be retrieved by inspecting the [image manifest](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/manifest.md) of the artifact. For artifacts stored locally as an [OCI Image layout](https://github.com/opencontainers/image-spec/blob/v1.0.0/image-layout.md), the descriptor info can be retrieved from the `index.json` file from OCI Image layout. Users can also pass descriptor info as a JSON file to Notation directly.
```

Comment:
`Users can also pass descriptor info as a JSON file to Notation directly.` The corresponding implementation is not in Notation as of today. Should we include this in our Notation 1.0 scope?

Reply:
It's not in 1.0 scope. I think I should remove it for now.